engine/rpcserver.go

```diff
@@ -804,7 +804,13 @@ func (s *RPCServer) SubmitOrder(ctx context.Context, r *gctrpc.SubmitOrderReques
 		Amount:   r.Amount,
 		Price:    r.Price,
 		ClientID: r.ClientId,
+		Exchange: r.Exchange,
 	})
+
+	if err != nil {
+		return &gctrpc.SubmitOrderResponse{}, err
+	}
+
```

Comment:
Here, I also suggest merging it into the master branch. It would panic if &gctrpc.SubmitOrderResponse is nil